posthog/api/authentication.py

```diff
@@ -337,3 +342,14 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         response = super().retrieve(request, *args, **kwargs)
         response.status_code = self.SUCCESS_STATUS_CODE
         return response
+
+
+class PasswordResetTokenGenerator(PasswordResetTokenGenerator):
+    def _make_hash_value(self, user: AbstractBaseUser, timestamp):
+        # Due to type differences between the user model and the token generator, we need to
+        # re-fetch the user from the database to get the correct type.
+        usable_user: User = User.objects.get(pk=user.pk)
```

Comment:
Liskov substition principle, I know – but this just seems wasteful. Can we bend the principle and instead throw an error if `not isinstance(user, User)`?

Reply:
So I tried this but it caused the token check to fail. I think it might be because the `user` that comes into this function doesn't have the `requested_password_reset_at` field, so even if it satisfies our User type, the data is missing. So back to my original implementation I guess, though it was a good suggestion.